src/mscorlib/src/Internal/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -94,117 +107,249 @@ public static int SizeOf<T>()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static ref T Add<T>(ref T source, int elementOffset)
         {
+#if PROJECTN
+            return ref AddByteOffset(ref source, (IntPtr)(elementOffset * (nint)SizeOf<T>()));
+#else
             // The body of this function will be replaced by the EE with unsafe code!!!
             // See getILIntrinsicImplementationForUnsafe for how this happens.
             typeof(T).ToString(); // Type token used by the actual method body
-            throw new InvalidOperationException();
+            throw new PlatformNotSupportedException();
+#endif
         }
 
         /// <summary>
-        /// Adds an element offset to the given reference.
+        /// Adds an element offset to the given pointer.
         /// </summary>
-        [CLSCompliant(false)]
         [NonVersionable]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static void* Add<T>(void* source, int elementOffset)
         {
+#if PROJECTN
+            return (byte*)source + (elementOffset * (nint)SizeOf<T>());
+#else
             // The body of this function will be replaced by the EE with unsafe code!!!
             // See getILIntrinsicImplementationForUnsafe for how this happens.
             typeof(T).ToString(); // Type token used by the actual method body
-            throw new InvalidOperationException();
-        }
-
-        /// <summary>
-        /// Determines the byte offset from origin to target from the given references.
-        /// </summary>
-        [NonVersionable]
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static IntPtr ByteOffset<T>(ref T origin, ref T target)
-        {
-            // The body of this function will be replaced by the EE with unsafe code!!!
-            // See getILIntrinsicImplementationForUnsafe for how this happens.
-            throw new InvalidOperationException();
+            throw new PlatformNotSupportedException();
+#endif
         }
 
         /// <summary>
         /// Adds an element offset to the given reference.
         /// </summary>
-        [CLSCompliant(false)]
+        [Intrinsic]
         [NonVersionable]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static ref T AddByteOffset<T>(ref T source, nuint byteOffset)
```

Comment:
This method is outlier - it is not in the official S.R.CS.Unsafe. It should stay internal. We want the public surface to be the same between 32-bit and 64-bit versions. nuint is changing between 32-bit and 64-bit versions.